Mathlib/RingTheory/KrullDimension/Zero.lean

```diff
@@ -0,0 +1,230 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.Jacobson.Ring
+import Mathlib.RingTheory.Spectrum.Prime.Topology
+
+/-!
+
+# Zero-dimensional rings
+
+We provide further API for zero-dimensional rings.
+Basic definitions and lemmas are provided in `Mathlib/RingTheory/KrullDimension/Basic.lean`.
+
+-/
+
+section CommSemiring
+
+variable {R : Type*} [CommSemiring R] [Ring.KrullDimLE 0 R] (I : Ideal R)
+
+lemma Ring.KrullDimLE.mem_minimalPrimes_iff {I J : Ideal R} :
+    I ∈ J.minimalPrimes ↔ I.IsPrime ∧ J ≤ I :=
+  ⟨fun H ↦ H.1, fun H ↦ ⟨H, fun _ h e ↦ (h.1.isMaximal'.eq_of_le H.1.ne_top e).ge⟩⟩
+
+lemma Ring.KrullDimLE.mem_minimalPrimes_iff_le_of_isPrime {I J : Ideal R} [I.IsPrime] :
+    I ∈ J.minimalPrimes ↔ J ≤ I := by
+  rwa [mem_minimalPrimes_iff, and_iff_right]
+
+variable (R) in
+lemma Ring.KrullDimLE.minimalPrimes_eq_setOf_isPrime :
+    minimalPrimes R = { I | I.IsPrime } := by
+  ext; simp [minimalPrimes, mem_minimalPrimes_iff]
+
+variable (R) in
+lemma Ring.KrullDimLE.minimalPrimes_eq_setOf_isMaximal :
+    minimalPrimes R = { I | I.IsMaximal } := by
+  ext; simp [minimalPrimes_eq_setOf_isPrime, Ideal.isMaximal_iff_isPrime]
+
+/-- Note that the `ringKrullDim` of the trivial ring is `⊥` and not `0`. -/
+instance (priority := 100) [Subsingleton R] : Ring.KrullDimLE 0 R :=
+  ⟨show ringKrullDim R ≤ 0 by rw [ringKrullDim_eq_bot_of_subsingleton]; exact bot_le⟩
+
+lemma Ring.KrullDimLE.isField_of_isDomain [IsDomain R] : IsField R := by
+  by_contra h
+  obtain ⟨p, hp, h⟩ := Ring.not_isField_iff_exists_prime.mp h
+  exact hp.symm (Ideal.bot_prime.isMaximal'.eq_of_le h.ne_top bot_le)
+
+section IsLocalRing
+
+omit [Ring.KrullDimLE 0 R] in
+variable (R) in
+lemma Ring.krullDimLE_zero_and_isLocalRing_tfae :
+    List.TFAE
+    [ Ring.KrullDimLE 0 R ∧ IsLocalRing R,
+      ∃! I : Ideal R, I.IsPrime,
+      ∀ x : R, IsNilpotent x ↔ ¬ IsUnit x,
+      (nilradical R).IsMaximal ] := by
```

Comment:
We could add `Subsingleton (PrimeSpectrum R)` into the mix.

Reply:
You probably need `Nonempty (Unique (PrimeSpectrum R))` instead. I opt to not add it for now as I think `TFAE` shouldn't be end results (due to the poor elaboration problem above) but rather intermediate lemmas. And it is easier to add `Ring.KrullDimLE.subsingleton_primeSpectrum` directly.

Reply:
Indeed, I neglected nonemptiness. Nonempty ∧ Subsingleton would also work.